pkgs/development/compilers/edk2/default.nix

```diff
@@ -12,7 +25,17 @@ else if stdenv.isAarch64 then
 else
   throw "Unsupported architecture";
 
-edk2 = stdenv.mkDerivation {
+buildStdenv = if stdenv.isDarwin then
```

Comment:
```suggestion
buildStdenv = if stdenv.isClang then
```

Reply:
Reason is we might want to support FreeBSD at some point too.

Reply:
That would suggest that the clang build would work on Linux if `stdenv` is overridden with something using clang, which is not the case. Because of the edk2 build system, clang only works on macOS.

Reply:
Ok. Let's leave it like that: https://github.com/tianocore/edk2/blob/d19040804afb2bdd60f18e8aef7da78028575fe6/BaseTools/Source/C/Makefiles/header.makefile#L70